serializers/src/glTF/2.0/babylon.glTFMaterial.ts

```diff
@@ -419,22 +363,18 @@ module BABYLON.GLTF2 {
                     const scene = texture.getScene();
                     if (scene) {
                         const proceduralTexture = new ProceduralTexture('texture', texture.getSize(), 'setAlphaToOne', scene);
-                        
-                        if (proceduralTexture) {
-                            proceduralTexture.setTexture('textureSampler', texture as Texture);
-                            proceduralTexture.onLoadObservable.add(() => { resolve(proceduralTexture) });
-                        }
-                        else {
-                            reject(`Cannot create procedural texture for ${texture.name}!`);
-                        }
+
+                        proceduralTexture.setTexture('textureSampler', texture as Texture);

```

Comment:
There are many casts from `BaseTexture` to `Texture` in this function. Are these all necessary? Perhaps the function should take a `Texture` instead of `BaseTexture`?

Reply:
@bghgary I just refactored to use `Texture` instead